.github/workflows/ci.yaml

```diff
@@ -0,0 +1,316 @@
+      - run: helm lint deployments/gpu-operator/
+  validate-csv:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Get Golang version
+        id: vars
+        run: |
+          GOLANG_VERSION=$( grep "GOLANG_VERSION ?=" versions.mk )
+          echo "GOLANG_VERSION=${GOLANG_VERSION##GOLANG_VERSION ?= }" >> $GITHUB_ENV
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GOLANG_VERSION }}
+      - run: make validate-csv
+  validate-helm-values:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Get Golang version
+        id: vars
+        run: |
+          GOLANG_VERSION=$( grep "GOLANG_VERSION ?=" versions.mk )
+          echo "GOLANG_VERSION=${GOLANG_VERSION##GOLANG_VERSION ?= }" >> $GITHUB_ENV
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GOLANG_VERSION }}
+      - run: make validate-helm-values
+
+  ### Golang checks and build ###
+  go-check:
+    needs: [helm-lint, validate-csv, validate-helm-values]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        name: Checkout code
+      - name: Get Golang version
+        id: vars
+        run: |
+          GOLANG_VERSION=$( grep "GOLANG_VERSION ?=" versions.mk )
+          echo "GOLANG_VERSION=${GOLANG_VERSION##GOLANG_VERSION ?= }" >> $GITHUB_ENV
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GOLANG_VERSION }}
+      - name: Lint
+        uses: golangci/golangci-lint-action@v5
+        with:
+          version: latest
+          args: -v --timeout 5m
+          skip-cache: true
+      - run: make check
+  go-test:
+    needs: [helm-lint, validate-csv, validate-helm-values]
+    name: unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Get Golang version
+        id: vars
+        run: |
+          GOLANG_VERSION=$( grep "GOLANG_VERSION ?=" versions.mk )
+          echo "GOLANG_VERSION=${GOLANG_VERSION##GOLANG_VERSION ?= }" >> $GITHUB_ENV
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GOLANG_VERSION }}
+      - run: make coverage
+  go-build:
+    needs: [helm-lint, validate-csv, validate-helm-values]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        name: Checkout code
+      - run: make docker-build
+
+  ### Image builds ###
+  build-gpu-operator:
+    needs: [go-check, go-test, go-build]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        dist: [ubi8]
+    steps:
+      - uses: actions/checkout@v4
+        name: Check out code
+      - name: Calculate build vars
+        id: vars
+        run: |
+          echo "COMMIT_SHORT_SHA=${GITHUB_SHA:0:8}" >> $GITHUB_ENV
```

Comment:
Can we move this to the top level `env` field too?

Also I prefer `GITHUB_SHORT_SHA` over `COMMIT_SHORT_SHA`

Reply:
We are leveraging shell [paramater expansion ](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) to extract the short sha, so  I don't believe we can define this in the top level `env` field, in which we cannot leverage a shell-like syntax.

Reply:
Sure, agreed that the `env` field is not an option.

Can we rename `COMMIT_SHORT_SHA` to `GITHUB_SHORT_SHA`?